src/connectors/menu/connectMenu.js

```diff
@@ -233,9 +233,20 @@ export default function connectMenu(renderFn, unmountFn) {
       dispose({ state }) {
         unmountFn();
 
-        const nextState = state
-          .removeHierarchicalFacetRefinement(attributeName)
-          .removeHierarchicalFacet(attributeName);
+        let nextState;
+
+        if (state.isHierarchicalFacetRefined(attributeName)) {
+          nextState = state.removeHierarchicalFacetRefinement(attributeName);
+        }
+
+        nextState = (nextState || state).removeHierarchicalFacet(attributeName);
```

Comment:
If you initialize `nextState` with `state` at l.236 you don't need to use `nextState || state`.